cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -211,6 +211,32 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 			infraSpec.StorageAccount = &scaffold.StorageAccount{
 				Containers: props.Containers,
 			}
+		case ResourceTypeAiModel:
+			props, castOk := res.Props.(AiFoundryModelProps)
```

Comment:
It's okay to forcefully panic here. The only way we would land here is that the marshal/unmarshal in `resources.go` was not done right.